compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/types/IrTypeSystemContext.kt

```diff
@@ -347,7 +347,7 @@ interface IrTypeSystemContext : TypeSystemContext, TypeSystemCommonSuperTypesCon
     override fun TypeConstructorMarker.getPrimitiveType(): PrimitiveType? {
         if (this !is IrClassSymbol || !isPublicApi) return null
 
-        val signature = signature.asPublic()
+        val signature = signature?.asPublic()
```

Comment:
The call to `isPublicApi` above is no longer needed here and in `getPrimitiveArrayType`